ocs_ci/deployment/baremetal.py

```diff
@@ -1109,6 +1112,63 @@ def deploy(self, log_cli_level="DEBUG"):
                         ignore_error=True,
                     )
 
+        def pending_user_action_handler(self, host):
+            """
+            Method for handling pending user action during deployment (this usually means that the server didn't boot
+            properly from the disk.)
+
+            Args:
+                host (dict): details about host with pending user action (from Assisted Installer api)
+
+            """
+            machine = host["requested_hostname"]
+            # do not run the handler for particular machine more often than once in 20 minutes
+            # to make sure the server have time to reboot and connect to the cluster
+            if (
+                time.time() - self.handled_pending_user_actions.get(machine, 0)
+                < 20 * 60
+            ):
+                logger.info(
+                    f"Skipping handling pending user action for {machine}. "
+                    "It was already handled less than 20 minutes ago."
+                )
+                return
+
+            if self.srv_details[machine].get("mgmt_provider", "ipmitool") == "ipmitool":
+                secrets = [
+                    self.srv_details[machine]["mgmt_username"],
+                    self.srv_details[machine]["mgmt_password"],
+                ]
+                # reboot machine
+                cmd = (
+                    f"ipmitool -I lanplus -U {self.srv_details[machine]['mgmt_username']} "
+                    f"-P {self.srv_details[machine]['mgmt_password']} "
+                    f"-H {self.srv_details[machine]['mgmt_console']} chassis power cycle || "
+                    f"ipmitool -I lanplus -U {self.srv_details[machine]['mgmt_username']} "
+                    f"-P {self.srv_details[machine]['mgmt_password']} "
+                    f"-H {self.srv_details[machine]['mgmt_console']} chassis power on"
+                )
+                rc, stdout, stderr = self.helper_node_handler.exec_cmd(
+                    cmd=cmd, secrets=secrets
+                )
+                assert (
+                    rc == 0
+                ), f"Command execution failed - rc: {rc}, stdout: '{stdout}', stderr: '{stderr}'"
+
+            elif (
+                self.srv_details[machine].get("mgmt_provider", "ipmitool") == "ibmcloud"
+            ):
+                ibmcloud = ibmcloud_bm.IBMCloudBM()
+                m = ibmcloud.get_machines_by_names([machine])
```

Comment:
I am not sure if we support AI now only on IBM cloud - but this function seems like BM on IBM Cloud related - do we conditionalize it somehow to run only on this kind of platform on IBM Cloud? And maybe worth to rename this function to be more telling it's only for ibm cloud.

Reply:
Ok, so let's go ahead with merge ;)